### PR TITLE
OX-6075: Support for FTP only users

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/storage/layer3/VirtualFileSystem.java [hunk 1]
```diff
@@ -29,6 +29,12 @@
 @Register(classes = VirtualFileSystem.class)
 public class VirtualFileSystem {
 
+    /**
+     * Defines the name of the sub scope used by the {@link sirius.biz.storage.layer3.downlink.ftp.FTPServer} and
+     * {@link sirius.biz.storage.layer3.downlink.ssh.SSHServer} which grant access per FTP, SFTP or SCP.
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/storage/layer3/VirtualFileSystem.java:34]:
```suggestion
     * {@link sirius.biz.storage.layer3.downlink.ssh.SSHServer} which grants access per FTP, SFTP or SCP.
```

File: src/main/java/sirius/biz/storage/layer3/downlink/ftp/BridgeUserManager.java [hunk 2]
```diff
@@ -94,6 +95,11 @@ public BridgeUser authenticate(Authentication authentication) throws Authenticat
                 StorageUtils.LOG.FINE("Layer 3/FTP: Invalid credentails...");
                 throw new AuthenticationFailedException("Invalid credentials.");
             }
+
+            if (!authUser.isSubScopeEnabled(VirtualFileSystem.SUB_SCOPE_VFS)) {
+                throw new AuthenticationFailedException("The required sub scope is not enabled for this use.");
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/storage/layer3/downlink/ftp/BridgeUserManager.java:100]:
```suggestion
                throw new AuthenticationFailedException("The required sub scope is not enabled for this user.");
```

File: src/main/java/sirius/biz/storage/layer3/downlink/ssh/SSHServer.java [hunk 3]
```diff
@@ -137,25 +137,34 @@ private boolean authenticate(String username, String password, ServerSession ses
...
         }
 
         UserInfo authUser = UserContext.get().getUserManager().findUserByCredentials(null, username, password);
+
+        if (authUser == null) {
+            return false;
+        }
+
+        if (!authUser.isSubScopeEnabled(VirtualFileSystem.SUB_SCOPE_VFS)) {
+            StorageUtils.LOG.FINE("Layer 3/SSH: The required sub scope is not enabled for: " + username);
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/storage/layer3/downlink/ssh/SSHServer.java:157]: Maybe log on Info?

**Review by mkeckmkeck — APPROVED** [review]

File: src/main/java/sirius/biz/storage/layer3/downlink/ssh/SSHServer.java (hunk 3 above)
Comment (andyHa) [reply src/main/java/sirius/biz/storage/layer3/downlink/ssh/SSHServer.java:157]: all other auth relevant messages are FINE as well therefore I used the same level here...

**Review by idlira — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Pull request merged**